hack/dind

```diff
@@ -27,10 +27,13 @@ fi
 
 # cgroup v2: enable nesting
 if [ -f /sys/fs/cgroup/cgroup.controllers ]; then
-	# move the init process (PID 1) from the root group to the /init group,
+	# move the processes from the root group to the /init group,
 	# otherwise writing subtree_control fails with EBUSY.
 	mkdir -p /sys/fs/cgroup/init
-	echo 1 > /sys/fs/cgroup/init/cgroup.procs
+	while read -r pid; do
+		# Ignore an error during moving non-existent process (i.e., "cat")
+		echo "${pid}" > /sys/fs/cgroup/init/cgroup.procs 2>&1 || true
+	done < /sys/fs/cgroup/cgroup.procs
```

Comment:
Nice! :smile:

Doing a little code golf, I was able to have this work successfully with something like this: :innocent:

```suggestion
	xargs -rn1 < /sys/fs/cgroup/cgroup.procs > /sys/fs/cgroup/init/cgroup.procs || :
```

Even starting a bunch of extra shell processes, it still seems to work properly:

```console
$ docker run -it --rm --name test --init --privileged docker:dind sh
/ # sh
/ # sh
/ # sh
/ # ps faux
PID   USER     TIME  COMMAND
    1 root      0:00 /sbin/docker-init -- dockerd-entrypoint.sh sh
    8 root      0:00 sh
   27 root      0:00 sh
   28 root      0:00 sh
   29 root      0:00 sh
   30 root      0:00 ps faux
/ # mkdir -p /sys/fs/cgroup/init
/ # xargs -rn1 < /sys/fs/cgroup/cgroup.procs > /sys/fs/cgroup/init/cgroup.procs || :
/ # sed -e 's/ / +/g' -e 's/^/+/' < /sys/fs/cgroup/cgroup.controllers > /sys/fs/cgroup/cgroup.subtree_control
/ # cat /sys/fs/cgroup/init/cgroup.procs
1
8
27
28
29
40
```

Reply:
updated, thanks